### PR TITLE
Tidy up UNL module:

### PR DESCRIPTION
**Pull request by vinniefalco**
* Renamed module to unl
* Renamed classes and members
* Removed cyclic dependency in Horizon

### CODE REVIEW

**Comment by nbougalis** [comment]
:+1:

File: src/ripple/unl/UNLManager.h [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_VALIDATORS_MANAGER_H_INCLUDED
```
Comment (scottschurr) [inline src/ripple/unl/UNLManager.h]: I'm guessing this include guard wants to change.

**Comment by scottschurr** [comment]
I gotta say, in terms of naming, "Module" was not very descriptive but "unl" may be worse.  I'm currently guessing that "unl" stands for "unique node list"?  I'm making that guess because I found the phrase "unique node list" buried on a single line in the change list.  (https://github.com/ripple/rippled/pull/1194/files#diff-ce7d72417f2df378ada1a52a94396f56R110) I found no other clue to what the term "unl" might mean.

Please at least document what the acronym means.

If "unl" does indeed mean "unique node list", this is still (in my opinion) a poor choice.  In my experience with the rippled code base then term "node" is primarily associated with the SHAMap.  The code affected by this change seems to be primarily about tracking validators.  The README.md (https://github.com/vinniefalco/rippled/blob/unl/src/ripple/unl/README.md) reinforces this impression.  So "node" strikes me as a real misdirection.

If we're tracking validators, it seems to me the name should include something about validators.  Perhaps this code should be in the pre-existing "src/ripple/validators" subtree, as suggested by the README.md.  Or perhaps "ChosenValidators" (https://github.com/ripple/rippled/pull/1194/files#diff-ce7d72417f2df378ada1a52a94396f56R33) would be a good choice.

**Comment by vinniefalco** [comment]
1. The concept of UNL is central to Ripple
2. "Unique Node List" is already so pervasive in the literature and papers that changing it is not practical
3. The documentation comment you're looking at is not final
4. New code to manage the UNL and the overlay is still in the process of being designed and has not been fully defined or written.
5. The unl module is experimental only and should not be used for production - I should have made this more clear in the pull request:

*Reviewers should ensure that the UNL code is not called in production code paths*

File: src/ripple/overlay/impl/PeerImp.cpp [hunk 2]
```diff
@@ -423,7 +424,10 @@ PeerImp::close()
 void
 PeerImp::fail(std::string const& reason)
 {
-    assert(strand_.running_in_this_thread());
+    if(! strand_.running_in_this_thread())
+        return strand_.post(std::bind (
+            (void(Peer::*)(std::string const&))&PeerImp::fail,
+                shared_from_this(), reason));
```
Comment (scottschurr) [inline src/ripple/overlay/impl/PeerImp.cpp]: My ignorance of the inner workings of `asio` is getting in my way here.  I'm hoping that we're very confident that the call to `strand_.post` makes a copy of `reason`.

File: src/ripple/unl/impl/Logic.h [hunk 3]
```diff
@@ -24,8 +24,9 @@
 #include <ripple/basics/hardened_hash.h>
```
Comment (scottschurr) [inline src/ripple/unl/impl/Logic.h]: The include guard name on line 20 should probably be updated.

File: src/ripple/overlay/impl/PeerImp.cpp (hunk 2 above)
Comment (vinniefalco) [inline src/ripple/overlay/impl/PeerImp.cpp]: Its got nothing to do with the behavior of `post` and everything to do with the behavior of `bind`. And `bind` always takes arguments by value.

File: src/ripple/unl/make_Manager.h [hunk 4]
```diff
@@ -20,7 +20,7 @@
 #ifndef RIPPLE_VALIDATORS_MAKE_MANAGER_H_INCLUDED
```
Comment (scottschurr) [inline src/ripple/unl/make_Manager.h]: Update include guard?

File: src/ripple/unl/impl/Tuning.h [hunk 5]
```diff
@@ -21,7 +21,7 @@
 #define RIPPLE_VALIDATORS_TUNING_H_INCLUDED
```
Comment (scottschurr) [inline src/ripple/unl/impl/Tuning.h]: Update include guard?

File: src/ripple/unl/impl/StoreSqdb.h [hunk 6]
```diff
@@ -20,13 +20,13 @@
 #ifndef RIPPLE_VALIDATORS_STORESQDB_H_INCLUDED
 #define RIPPLE_VALIDATORS_STORESQDB_H_INCLUDED
```
Comment (scottschurr) [inline src/ripple/unl/impl/StoreSqdb.h]: Update include guard?

File: src/ripple/unl/impl/Store.h [hunk 7]
```diff
@@ -21,7 +21,7 @@
 #define RIPPLE_VALIDATORS_STORE_H_INCLUDED
```
Comment (scottschurr) [inline src/ripple/unl/impl/Store.h]: Update include guard?

File: src/ripple/overlay/impl/PeerImp.cpp (hunk 2 above)
Comment (scottschurr) [inline src/ripple/overlay/impl/PeerImp.cpp]: Thanks.

**Comment by scottschurr** [comment]
Assuming the include guards are updated, :+1:.

**Comment by vinniefalco** [comment]
Part of #1194

**Pull request closed without merging**